### PR TITLE
Add ERR_raise() errors to fips OSSL_provider_init and self tests.

### PR DESCRIPTION
**Pull request by slontis**
As the ERR_raise() is setup in OSSL_provider_init  returning a range of of negative values for errors is not required. This will need to be revisited if the code ever moves to running from the DEP.

Added a -config option to the fips install so that it can test if a fips module is loadable from configuration. (The -verify option only uses the generated config, whereas -config uses the normal way of including the generated data via another config file).
    
Added more failure tests for the raised errors.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: doc/man1/openssl-fipsinstall.pod.in [hunk 1]
```diff
@@ -143,6 +144,14 @@ Either option or both may be used to select the tests to corrupt.
 Refer to the entries for B<st-desc> and B<st-type> in L<OSSL_PROVIDER-FIPS(7)> for
 values that can be used.
 
+=item B<-config> I<parent_config>
+
+Test that a FIPS provider can be loaded from configuration data.
```
Comment (richsalz) [inline doc/man1/openssl-fipsinstall.pod.in:149]: perhaps "from the specified configuration file"

File: doc/man1/openssl-fipsinstall.pod.in [hunk 2]
```diff
@@ -143,6 +144,14 @@ Either option or both may be used to select the tests to corrupt.
...
+A previous call to this application needs to generate the extra configuration
+data that is included by the base <parent_config> configuration file,
```
Comment (richsalz) [inline doc/man1/openssl-fipsinstall.pod.in:151]: Need markup before the angle brackets, as in ```C<parent_config>```

File: doc/man1/openssl-fipsinstall.pod.in [hunk 3]
```diff
@@ -143,6 +144,14 @@ Either option or both may be used to select the tests to corrupt.
...
+See L<config(5)> for further information on how to set up a provider section.
```
Comment (richsalz) [inline doc/man1/openssl-fipsinstall.pod.in:152]: lowercase the "See" or make the comma above a full-stop.

File: apps/fipsinstall.c [hunk 4]
```diff
@@ -343,6 +352,17 @@ int fipsinstall_main(int argc, char **argv)
...
     }
     argc = opt_num_rest();
+
+    if (parent_config != NULL) {
+        /* Test that a parent config can load the module */
+        if (verify_module_load(parent_config)) {
+            ret = OSSL_PROVIDER_available(NULL, "fips") ? 0 : 1;
+            if (!quiet)
+                BIO_printf(bio_out, "FIPS provider is %s\n",
+                           ret == 0 ? "Available" :" Not Available");
```
Comment (paulidale) [inline apps/fipsinstall.c:362]: `s/:" /: "/`

Comment (richsalz) [reply apps/fipsinstall.c:362]: And remove the uppercase.

File: apps/fipsinstall.c [hunk 5]
```diff
@@ -64,6 +64,7 @@ const OPTIONS fipsinstall_options[] = {
     {"noout", OPT_NO_LOG, '-', "Disable logging of self test events"},
     {"corrupt_desc", OPT_CORRUPT_DESC, 's', "Corrupt a self test by description"},
     {"corrupt_type", OPT_CORRUPT_TYPE, 's', "Corrupt a self test by type"},
+    { "config", OPT_CONFIG, '<', "The parent config to verify"},
```
Comment (mattcaswell) [inline apps/fipsinstall.c:67]: Nit: there seems to be a spurious extra space after the open brace on this line, compared to the other lines.

**Comment by slontis** [comment]
rebased and sfixed space nit - should not need a reapproval.

**Comment by slontis** [comment]
rebased again to fix merge conflicts and to update the generated files..
Should not need reapproval.

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**